app/react/I18N/components/I18NMenu.js

```diff
@@ -15,8 +15,12 @@ class I18NMenu extends Component {
 
   render() {
     const { languages, locale, location, i18nmode, toggleInlineEdit } = this.props;
-
     let path = location.pathname;
+
+    if (location.search.match(/page=/)) {
```

Comment:
Are you sure we don't use "page" in other type of navigation properties like pagination?  Please check that as this could really affect this approach.

Reply:
I understand this component I18NMenu is only used to construct the language navigation menu, so other uses of the page query should not be affected.

Reply:
Yes, pagination for any URL that could get affected by this regexp: 
- IX is one good example
- activity log
- library "load more"

This that I can "think of" don't use the Page, so it seems like it should work without affecting other places.  Maybe we can brainstorm other places that could use this?